beater/api/mux.go

```diff
@@ -131,26 +128,22 @@ func rumHandler(cfg *config.Config, reporter publish.Reporter) (request.Handler,
 	if err != nil {
 		return nil, err
 	}
-	cache, err := intake.NewRateLimitLRUCache(cfg.RumConfig.EventRate.LruSize, cfg.RumConfig.EventRate.Limit, burstMultiplier)
-	if err != nil {
-		return nil, err
-	}
 	h := intake.Handler(userMetaDataDecoder(cfg, emptyDecoder),
 		&stream.Processor{
 			Tconfig:      *tcfg,
 			Mconfig:      model.Config{Experimental: cfg.Mode == config.ModeExperimental},
 			MaxEventSize: cfg.MaxEventSize,
 		},
-		cache,
 		reporter)
-	return middleware.Wrap(h, rumMiddleware(cfg)...), nil
+	return middleware.Wrap(h, rumMiddleware(cfg)...)
 }
 
 func rumMiddleware(cfg *config.Config) []middleware.Middleware {
 	return append(apmMiddleware(intake.MonitoringMap),
 		middleware.KillSwitchMiddleware(cfg.RumConfig.IsEnabled()),
 		middleware.RequestTimeMiddleware(),
-		middleware.CORSMiddleware(cfg.RumConfig.AllowOrigins))
+		middleware.CORSMiddleware(cfg.RumConfig.AllowOrigins),
+		middleware.SetRateLimitMiddleware(cfg.RumConfig.EventRate))
```

Comment:
Nit: I'd put the rate limit middleware right behind the killswitch one. In this case it doesn't make a difference since none of the other middlewares use it, but that should be an implementation detail.

Reply:
out of curiosity, if it doesn't make a difference, why higher in the list is better?

Reply:
IMO: Theoretically a rate limiter should be set as soon as possible, so it can be applied as soon as possible. Now the way we use it in the Intake API is special, so it doesn't make a difference. But what if the `CORSMiddleware` used it for some check already to not even handle a preflight request if rate limit is exceeded? 

I know, it's not important now, as it is just theoretical.. that's why I said _NIT_ and didn't require to change..